.github/workflows/production_on_push.yml

```diff
@@ -0,0 +1,82 @@
+name: production_on_push
+
+on:
+  push:
+    branches:
+      - main
+
+jobs:
+  sonarqube:
+    name: analyze
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          # Disabling shallow clone is recommended for improving relevancy of reporting
+          fetch-depth: 0
+      
+      - name: sonarqube-scan-branch
+        uses: sonarsource/sonarqube-scan-action@master
+        env:
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+          SONAR_HOST_URL: ${{ secrets.SONAR_HOST_URL }}
+        with:
+          args: >
+            -Dsonar.branch.name=${{ github.event.ref }}
+
+  
+  build:
+    runs-on: ubuntu-latest
+    outputs:
+      image_ref: ${{ vars.DOCKER_REGISTRY }}/${{ vars.DOCKER_USERNAME }}/boilerplate-react-native@${{ steps.build.outputs.image_digest }}
+    steps:
+      - name: Checkout (app)
+        uses: actions/checkout@v3
+        # make sure to keep checked in code at different directory to avoid colliding with docker caching
+        with:
+          path: app
+
+      - name: Extract branch name
+        id: extract_branch
+        shell: bash
+        run: |
+          BRANCH_NAME=$(echo ${{ github.event.ref }} | sed -e 's/^refs\/heads\///g')
+          BRANCH_HASH=$(sha1sum < <(printf '%s' $BRANCH_NAME) | cut -c -15)
+
+          echo "##[set-output name=branch_hash;]$(echo $BRANCH_HASH)"
+
+      - name: Build application
+        id: build
+        uses: ./actions/build
```

Comment:
@codVenturer - Try out `./app/actions/build` instead. See line #41, your app is being checked out at `app`. It should fix the build issue you are facing.